### PR TITLE
Fixes to the navigation components

### LINKED ISSUE #1112: Input borders are cropped when wrapped with collapsible content using useCollapsible
## Describe the bug

Form elements overflow the container which uses styles provided by `useCollapsible` hook.

CleanShot 2021-08-13 at 10 39 04@2x

Please check [this sandbox](https://codesandbox.io/s/nervous-bardeen-x833d) for live example.

## Specifications

* Circuit version `3.0.1`

### PR DESCRIPTION
**Pull request by connor-baer**
Fixes #1112.

## Purpose

While integrating the new navigation components into an application, we've discovered a number of bugs.

## Approach and changes

- Filter non-HTML attributes in Popover
- Update the Popover position on open
- Prevent interactions with elements behind Popover overlay
- Export navigation types
- Fix the navigation's min-width when the secondary nav is hidden
- Close the mobile navigation modal when a link is clicked
- Only hide _vertical_ overflow in `useCollapsible`

## Definition of done

* [ ] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: .changeset/five-peaches-perform.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': minor
```
Comment (robinmetral) [inline .changeset/five-peaches-perform.md:2]: Patch?

**Review by robinmetral — APPROVED** [review]
Amazing! 🐛💥

**Review by amelako — APPROVED** [review]
🌟

File: .changeset/five-peaches-perform.md (hunk 1 above)
Comment (connor-baer) [reply .changeset/five-peaches-perform.md:2]:
```suggestion
'@sumup/circuit-ui': patch
```

**Pull request merged**